### ISSUE TITLE
Configure pre-commit hooks

### ISSUE THREAD

**Issue by algomaster99** [SpoonLabs/diffmin#86]
It's common to push a patch to realise the builds failed at the style check step eventually. Configuring pre-commit hooks would format the source code before committing the patch.

**Comment by slarse** [comment]
Pre-commit hooks are repository-local, you can't configure them for anyone but yourself. You can of course provide pre-commit hooks and instructions on how to install them, but there's no way to ensure that a given pre-commit hook is automatically installed when someone clones the repository.

**Comment by algomaster99** [comment]
> there's no way to ensure that a given pre-commit hook is automatically installed when someone clones the repository.

Yeah, I am aware of that but I find this design quite weird. Don't you think it would be better if `git` made sure that some hooks must be same for all users? If necessary, they could allow an option to extend it.

Anyway, JavaScript codebases have `husky` to install hooks. Do you have any suggestions for a similar package in Java?

**Comment by slarse** [comment]
> Don't you think it would be better if git made sure that some hooks must be same for all users?

Not really, because pre-commit hooks can be any script, and scripts are OS-dependent, while Git itself is multi-platform. To incorporate hooks into the general Git protocols, there'd need to be some Git-specific scripting language that works wherever Git works. And that's just not going to happen.

> Do you have any suggestions for a similar package in Java?

I don't know of anything like that for Java. The one I typically use is [pre-commit](https://pre-commit.com/), which requires Python 3.
